From 1d118c610ff3f123866ea4f393ed469f16d01fc2 Mon Sep 17 00:00:00 2001
From: B38613 <B38613@freescale.com>
Date: Fri, 20 Apr 2012 13:13:25 +0800
Subject: [PATCH 265/486] ENGR00180235 GPS: work not stable

intermittent report "could not open gps serial device
ttya0 EIO error" for 100 times,then it won't work.
Add usleep here, usually open ok in second time in this case.

Signed-off-by: Zhou Jianzheng <B38613@freescale.com>
---
 libgps/athr_gps.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libgps/athr_gps.c b/libgps/athr_gps.c
index fe0505b..b467081 100755
--- a/libgps/athr_gps.c
+++ b/libgps/athr_gps.c
@@ -1438,6 +1438,7 @@ gps_state_init( GpsState*  state )
      {
         retry_count++;
         state->fd = open( device, O_RDWR | O_NOCTTY | O_NONBLOCK);
+	usleep (100000);
      } while ((retry_count < 100) && state->fd < 0 && ((errno == EIO) || (errno == EINTR)));
 
     if (state->fd < 0)
-- 
1.8.0

